### ISSUE TITLE
"error: XFS: sizeof(struct xfs_efi_log_format_32) is wrong" in fs/xfs/xfs_super.c

### ISSUE THREAD

**Issue by GustavoARSilva** [KSPP/linux#57]
When replacing the one-element array **efi_extents[1]** with a flexible-array member in struct **xfs_efi_log_format_32**:

```
diff --git a/fs/xfs/libxfs/xfs_log_format.h b/fs/xfs/libxfs/xfs_log_format.h
index e3400c9c71cd..229784f79b0c 100644
--- a/fs/xfs/libxfs/xfs_log_format.h
+++ b/fs/xfs/libxfs/xfs_log_format.h
@@ -579,7 +579,7 @@ typedef struct xfs_efi_log_format_32 {
        uint16_t                efi_size;       /* size of this item */
        uint32_t                efi_nextents;   /* # extents to free */
        uint64_t                efi_id;         /* efi identifier */
-       xfs_extent_32_t         efi_extents[1]; /* array of extents to free */
+       xfs_extent_32_t         efi_extents[];  /* array of extents to free */
 } __attribute__((packed)) xfs_efi_log_format_32_t;
 
 typedef struct xfs_efi_log_format_64 {
```

the following **error** shows up:

```
                 from fs/xfs/xfs_super.c:7:
In function ‘xfs_check_ondisk_structs’,
    inlined from ‘init_xfs_fs’ at fs/xfs/xfs_super.c:2071:2:
./include/linux/compiler.h:394:38: error: call to ‘__compiletime_assert_1678’ declared with attribute error: XFS: sizeof(struct xfs_efi_log_format_32) is wrong, expected 28
  394 |  _compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
      |                                      ^
./include/linux/compiler.h:375:4: note: in definition of macro ‘__compiletime_assert’
  375 |    prefix ## suffix();    \
      |    ^~~~~~
./include/linux/compiler.h:394:2: note: in expansion of macro ‘_compiletime_assert’
  394 |  _compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
      |  ^~~~~~~~~~~~~~~~~~~
./include/linux/build_bug.h:39:37: note: in expansion of macro ‘compiletime_assert’
   39 | #define BUILD_BUG_ON_MSG(cond, msg) compiletime_assert(!(cond), msg)
      |                                     ^~~~~~~~~~~~~~~~~~
fs/xfs/xfs_ondisk.h:10:2: note: in expansion of macro ‘BUILD_BUG_ON_MSG’
   10 |  BUILD_BUG_ON_MSG(sizeof(structname) != (size), "XFS: sizeof(" \
      |  ^~~~~~~~~~~~~~~~
fs/xfs/xfs_ondisk.h:118:2: note: in expansion of macro ‘XFS_CHECK_STRUCT_SIZE’
  118 |  XFS_CHECK_STRUCT_SIZE(struct xfs_efi_log_format_32, 28);
      |  ^~~~~~~~~~~~~~~~~~~~~
scripts/Makefile.build:272: recipe for target 'fs/xfs/xfs_super.o' failed
```

See [flexible-array conversions](https://github.com/KSPP/linux/issues/21) issue.

**Comment by GustavoARSilva** [comment]
This patch should fix this issue:

https://git.kernel.org/pub/scm/linux/kernel/git/gustavoars/linux.git/commit/?h=testing/xfs

Waiting for it to be tested by the 0-day people (Kernel test robot lkp@intel.com).

**Comment by GustavoARSilva** [comment]
Last version: https://lore.kernel.org/lkml/20210311042302.GA137676@embeddedor/

Reviewed-by: Darrick J. Wong <djwong@kernel.org>

**Comment by GustavoARSilva** [comment]
Fixed by commit 03a7485cd701 ("xfs: fix memcpy fortify errors in EFI log format copying")
